diag_manager/diag_manager.F90

```diff
@@ -3945,9 +3945,11 @@ SUBROUTINE diag_manager_init(diag_model_subset, time_init, err_msg)
       ALLOCATE(input_fields(j)%output_fields(MAX_OUT_PER_IN_FIELD))
     END DO
     ALLOCATE(files(max_files))
+#ifndef use_mpp_io
```

Comment:
Are the `DEALLOCATE` statements somewhere for these items?

Reply:
`output_fields`, `input_fields`, `files`, `fileobjU`, `fileobj`, `fileobjND`, and `fnum_for_domain` are not deallocated anywhere that I can see.

Reply:
They really should be deallocated somewhere.

Reply:
I'm hesitant to do that because I don't know if anything will break.

Reply:
I added deallocate statements for the fms2_io variables that are in here